jadx-core/src/main/java/jadx/core/utils/StringUtils.java

```diff
@@ -193,4 +196,18 @@ private static void commonEscapeAndAppend(StringBuilder sb, char c) {
 			sb.append(c);
 		}
 	}
+
+	private static final Pattern P0 = Pattern.compile("^[0o]+$");
+	private static final Pattern P1 = Pattern.compile("^[iI]+$");
+	private static final Pattern P2 = Pattern.compile("^[uv]+$");
+	private static final Pattern P3 = Pattern.compile("[0-9]");
```

Comment:
I like the idea of implementing deobfuscation based on patterns.
But it is better to make it more general: store pattern and some options in list and process matching in loop